### PR TITLE
Bug 1882071: Fix bug in reflector not recovering from "Too large resource version"

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
updates the client-go and related kubernetes repos to v0.18.6

update the vendored client-go code to understand new error messages

in addition to `go mod vendor` to update the vendored dir ran
`go mod tidy`

Signed-off-by: Jacob Tanenbaum <jtanenba@redhat.com>

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@danwinship @p0lyn0mial updating the cluster-network-operators version of client-go

File: go.mod [hunk 1]
```diff
@@ -18,15 +19,15 @@ require (
...
+	k8s.io/api v0.18.6
+	k8s.io/apimachinery v0.18.6
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/kube-proxy v0.0.0-20190918162534-de037b596c1e
 	k8s.io/utils v0.0.0-20200603063816-c1c6865ac451
 	sigs.k8s.io/controller-runtime v0.5.2
 )
 
-replace k8s.io/client-go => k8s.io/client-go v0.18.3
+replace k8s.io/client-go => k8s.io/client-go v0.18.6
```
Comment (p0lyn0mial) [inline go.mod:30]: could you group replace into  ?
```
replace (
 entry
 entry
)
```

File: go.mod [hunk 2]
```diff
@@ -18,15 +19,15 @@ require (
...
 
 replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.3
```
Comment (p0lyn0mial) [inline go.mod:32]: why not to keep all your deps on the same level v0.18.6 ?

File: go.mod [hunk 3]
```diff
@@ -18,15 +19,15 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/vishvananda/netlink v1.0.0
 	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/api v0.18.4
-	k8s.io/apimachinery v0.18.4
+	k8s.io/api v0.18.6
+	k8s.io/apimachinery v0.18.6
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/kube-proxy v0.0.0-20190918162534-de037b596c1e
 	k8s.io/utils v0.0.0-20200603063816-c1c6865ac451
```
Comment (p0lyn0mial) [inline go.mod:26]: could you change it to v.0.18.6 as well ?

Comment (p0lyn0mial) [inline go.mod:26]: same for this, change it to v.0.18.6

Comment (p0lyn0mial) [reply go.mod:26]: and k8s.io/kube-proxy

File: go.mod (hunk 1 above)
Comment (JacobTanenbaum) [reply go.mod]: done

File: go.mod (hunk 3 above)
Comment (JacobTanenbaum) [reply go.mod:26]: cannot change k8s.io/utils to v.0.18.6 it doesnt version...I changed it to the version that kube 0.18.6 uses

Comment (JacobTanenbaum) [reply go.mod:26]: done

Comment (p0lyn0mial) [reply go.mod:26]: ic, no tag in that repo.

**Pull request merged**